custom/icds_reports/utils/aggregation_helpers/distributed/agg_awc.py

```diff
@@ -228,7 +227,7 @@ def updates(self):
             ) as cases_person_adolescent_girls_11_14_all,
             sum(
                 CASE WHEN %(month_end_11yr)s > dob AND %(month_start_14yr)s <= dob AND sex = 'F'
-                    AND migration_status IS DISTINCT FROM 1
+                    AND (agg_migration.is_migrated IS DISTINCT FROM 1 OR agg_migration.migration_date >= %(start_date)s)
```

Comment:
What is the intended behavior if the case is migrated part way through the month? On second glance at this expression it seems like the correct behavior should be `>= next_month_start_date` rather than the current month one. That could possibly also be written as `> end_date` but you would have to be careful about whether either value was a datetime rather than a date because `2020-01-01 12:00 > 2020-01-01` will return true even though you probably dont want it to. So that expression will only work if they are both dates (or you truncate one to a date)

Reply:
Still curious about this

Reply:
I not able to get what you actually want to ask, but in terms of logic things are correct and the date issues appears valid, added `::date` to avoid such issue

Reply:
I think I am just asking for a brief explanation in words who this is supposed to include and exclude? A general answer would be good, but I am also specifically interested in what the intended behavior for March data is someone is migrated on March 15. Without knowing what the intended behavior is, it is impossible to verify if the given logic is correct, but it appears to behave in a way that surprises me.

Reply:
Yeah, so as per the question, if a beneficiary is migrated on 15 march then he/she would be considered not-migrated in the march and migrated from April.